### DISCUSSION TITLE
git.Git.execute() does not work in v3.1.11

### DISCUSSION THREAD

**Discussion by zm0n3** [gitpython-developers/GitPython#1170] [Q&A] [answered]
When trying to execute any command by using the execute() method (e.g. `git.Git.execute("git config --global --list")` or `git.Git.execute(command="git config --global --list")`), this is the output I get:
> *** TypeError: execute() missing 1 required positional argument: 'self'

Is this a bug?

**Comment by Byron** [comment]
I recommend [reading the docs on how to use git directly](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly) to resolve this issue. The problem is trying to use an instance method as a class method.

**Comment by zm0n3** [comment]
> I recommend [reading the docs on how to use git directly](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly) to resolve this issue. The problem is trying to use an instance method as a class method.

Thanks Byron for pointing me the doc. So if I have to change the git global config, there's nothing I can do, am I correct?

**Comment by Byron** [comment]
You can manually instantiate a `Git` command [like so](https://gitpython.readthedocs.io/en/stable/reference.html#module-git.cmd).

**Comment by zm0n3** [comment]
> You can manually instantiate a `Git` command [like so](https://gitpython.readthedocs.io/en/stable/reference.html#module-git.cmd).

I was searching a way to manage the git config --global without having a repo setup. In particular, before cloning a repo I need to ensure that the **global** proxy settings are correctly placed in the config list (`git config --global http.proxy`, otherwise all the fetch/clone commands will fail), but apparently I cannot execute any git command if I do not either have a repo set up or a repo initialized.

A workaround could be to initialize a dummy empty repo which I can use to execute the global config commands, and then delete it. But it's a dirty solution... Do you have better ideas?

Thanks in advance

**Comment by Byron** [comment, marked as answer]
```python
g = Git( git_dir )
rval = g.config(…)       
```

`git_dir` can be any existing directory, it is the working directory of the git command when spawned. This doesn't need to be a repository.
Maybe try it with the invocation you need.

**Comment by hhaoao** [comment]
This [example](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly) is too short, I re-written an example, hope it helps.

[for_each_ref](https://git-scm.com/docs/git-for-each-ref)


```python3
from git import Git

def update_force(path):
    """
        强制更新 
    """

    g = Git(path)
    g.config("--global", "http.proxy", "socks5://127.0.0.1:10808")
    g.checkout('*')
    pull_info = g.pull("--no-rebase")
    g.config("--global", "--unset", "http.proxy")
    info = g.for_each_ref()
    return info


```
